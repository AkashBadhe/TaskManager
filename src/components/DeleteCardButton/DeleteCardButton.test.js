import React from "react";
import { shallow } from "enzyme";
import DeleteCardButton from "./index";

describe("DeleteCardButton", () => {
  let deleteCardButton = shallow(<DeleteCardButton />);
  it("Should mount `DeleteCardButton`", () => {
    expect(deleteCardButton).toMatchSnapshot();
  });

  it("Should have `styled.button`", () => {
    expect(deleteCardButton.find("FaTimesCircle").exists()).toBe(true);
  });
});
