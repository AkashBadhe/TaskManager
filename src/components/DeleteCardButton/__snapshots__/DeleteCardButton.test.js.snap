// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteCardButton Should mount \`DeleteCardButton\` 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DeleteCardButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <FaTimesCircle
        size={16}
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "size": 16,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "baseHash": 400283751,
        "componentId": "sc-AxjAm",
        "isStatic": false,
        "rules": Array [
          "
  padding: 0;
  position: relative;
  border: 0;
  opacity: 0.1;
  cursor: pointer;
  display: flex;
  justify-content: center;
  align-items: center;
  color: rgb(46, 68, 78);

  &:hover,
  &:focus,
  &:active {
    color: #da3849;
    opacity: 1;
  }
",
        ],
        "staticRulesId": "",
      },
      "displayName": "styled.button",
      "foldedComponentIds": Array [],
      "render": [Function],
      "shouldForwardProp": undefined,
      "styledComponentId": "sc-AxjAm",
      "target": "button",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <FaTimesCircle
          size={16}
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "size": 16,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "baseHash": 400283751,
          "componentId": "sc-AxjAm",
          "isStatic": false,
          "rules": Array [
            "
  padding: 0;
  position: relative;
  border: 0;
  opacity: 0.1;
  cursor: pointer;
  display: flex;
  justify-content: center;
  align-items: center;
  color: rgb(46, 68, 78);

  &:hover,
  &:focus,
  &:active {
    color: #da3849;
    opacity: 1;
  }
",
          ],
          "staticRulesId": "",
        },
        "displayName": "styled.button",
        "foldedComponentIds": Array [],
        "render": [Function],
        "shouldForwardProp": undefined,
        "styledComponentId": "sc-AxjAm",
        "target": "button",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
